windows/sound.cpp

```diff
@@ -109,8 +109,15 @@ strCurDevName = cDriverNames[iDriverIdx];
 void CSound::UnloadCurrentDriver()
 {
     // clean up ASIO stuff
-    ASIOStop();
-    ASIODisposeBuffers();
+    if ( bRun )
+    {
+        this->Stop();
+    }
+    if ( bufferInfos[0].buffers[0] )

```

Comment:
What happens here if `bufferInfos[0]` is unset?  (Can it get here in that situation?)

Reply:
I clear all of `bufferInfos` in the constructor to make sure that it has a defined value.

Reply:
This is ok, because it is addressing part of the struct, not following a pointer
(it's not `bufferInfos[0]->buffers[0]`)